### PR TITLE
Add Azure and GCP ingress access point and gateway support

### PR DESCRIPTION
**Pull request by ejagade**
Release Notes
-------------
New Features
- Added Azure Private Link ingress gateway and access point support (`--cloud azure --type ingress-privatelink`)
- Added GCP Private Service Connect ingress gateway and access point support (`--cloud gcp --type ingress-private-service-connect`)

Checklist
---------
- [x] I have successfully built and used a custom CLI binary, without linter issues from this PR.
- [x] I have clearly specified in the `What` section below whether this PR applies to Confluent Cloud, Confluent Platform, or both.
- [x] I have verified this PR in Confluent Cloud pre-prod or production environment, if applicable.
- [ ] I have verified this PR in Confluent Platform on-premises environment, if applicable.
- [x] I have attached manual CLI verification results or screenshots in the `Test & Review` section below.
- [x] I have added appropriate CLI integration or unit tests for any new or updated commands and functionality.
- [x] I confirm that this PR introduces no breaking changes or backward compatibility issues.
- [x] I have indicated the potential customer impact if something goes wrong in the `Blast Radius` section below.
- [x] I have put checkmarks below confirming that the feature associated with this PR is enabled in:
  - [x] Confluent Cloud prod
  - [x] Confluent Cloud stag
  - [ ] Confluent Platform
  - [ ] Check this box if the feature is enabled for certain organizations only

What
----
Adds Azure Private Link and GCP Private Service Connect ingress gateway and access point support to the CLI, completing multi-cloud coverage alongside the existing AWS ingress support.

- New gateway types: `ingress-privatelink` (Azure) and `ingress-private-service-connect` (GCP)
- New access point flags: `--private-endpoint-resource-id` (Azure), `--private-service-connect-connection-id` (GCP)
- Cross-cloud flag validation with clear error messages
- Updated list filter support for `azure-ingress-privatelink` and `gcp-ingress-private-service-connect`
- Applies to Confluent Cloud only.

Blast Radius
----
Confluent Cloud customers using `confluent network gateway` or `confluent network access-point private-link ingress-endpoint` commands on Azure or GCP may be impacted if something goes wrong. Existing AWS ingress commands are unchanged.

References
----------
- PR: https://github.com/confluentinc/cli/pull/3283

Test & Review
-------------
- [Staging Verification](https://confluentinc.atlassian.net/wiki/x/e4EnSwE) — all 15 manual tests passed on staging
- [Prod Sanity Check](https://confluentinc.atlassian.net/wiki/x/6oCxSwE) — all 10 tests passed in prod

### CODE REVIEW

File: internal/network/command_access_point_private_link_ingress_endpoint.go [hunk 1]
```diff
@@ -102,6 +109,23 @@ func printPrivateLinkIngressEndpointTable(cmd *cobra.Command, ingressEndpoint ne
...
+		out.AzurePrivateEndpointResourceId = ingressEndpoint.Status.Config.NetworkingV1AzureIngressPrivateLinkEndpointStatus.GetPrivateEndpointResourceId()
+		if ingressEndpoint.Status.Config.NetworkingV1AzureIngressPrivateLinkEndpointStatus.HasDnsDomain() {
+			out.DnsDomain = ingressEndpoint.Status.Config.NetworkingV1AzureIngressPrivateLinkEndpointStatus.GetDnsDomain()
+		}
+	}
+
+	if ingressEndpoint.Status.Config != nil && ingressEndpoint.Status.Config.NetworkingV1GcpIngressPrivateServiceConnectEndpointStatus != nil {
+		out.GcpPrivateServiceConnectServiceAttachment = ingressEndpoint.Status.Config.NetworkingV1GcpIngressPrivateServiceConnectEndpointStatus.GetPrivateServiceConnectServiceAttachment()
+		out.GcpPrivateServiceConnectConnectionId = ingressEndpoint.Status.Config.NetworkingV1GcpIngressPrivateServiceConnectEndpointStatus.GetPrivateServiceConnectConnectionId()
+		if ingressEndpoint.Status.Config.NetworkingV1GcpIngressPrivateServiceConnectEndpointStatus.HasDnsDomain() {
```
Comment (sgagniere) [inline internal/network/command_access_point_private_link_ingress_endpoint.go]: Nitpick: This guard is probably unnecessary since `GetDnsDomain` will return empty string if it's not set.

File: test/fixtures/output/network/access-point/private-link/ingress-endpoint/delete.golden [hunk 2]
```diff
@@ -1 +1,2 @@
+warning: GOCOVERDIR not set, no coverage data emitted
```
Comment (sgagniere) [inline test/fixtures/output/network/access-point/private-link/ingress-endpoint/delete.golden]: We'll want to remove these lines.

Running `make integration-test` should automatically export this var and remove this warning.

**Review by sgagniere — COMMENTED** [review]
Thanks for the PR! Here's a few comments:

File: internal/network/command_gateway.go [hunk 3]
```diff
@@ -17,42 +17,51 @@ import (
...
-		"azure-egress-privatelink":           azureEgressPrivateLink,
-		"gcp-egress-private-service-connect": gcpEgressPrivateServiceConnect,
+		"aws-egress-privatelink":              awsEgressPrivateLink,
+		"aws-ingress-privatelink":             awsIngressPrivateLink,
+		"azure-egress-privatelink":            azureEgressPrivateLink,
+		"azure-ingress-privatelink":           azureIngressPrivateLink,
+		"gcp-egress-privatelink":              "GcpEgressPrivateLink",
+		"gcp-ingress-privatelink":             "GcpIngressPrivateLink",
+		"gcp-egress-private-service-connect":  "GcpEgressPrivateLink",
+		"gcp-ingress-private-service-connect": "GcpIngressPrivateLink",
```
Comment (cqin-confluent) [inline internal/network/command_gateway.go]: These map values use a string the SDK doesn't define. Could yo uchange L41-42 to reference `gcpEgressPrivateServiceConnect` / `gcpIngressPrivateServiceConnect`? The [bug-bash GCP page](https://confluentinc.atlassian.net/wiki/spaces/~62bb4e90d752af0e54ec1975/pages/5292327700/CLI+test+for+GCP+IngressPrivateServiceConnect+Gateway+AP) already captured the API rejection: 
```
Error: Invalid gateway type: {[GcpIngressPrivateServiceConnect]}
```

File: internal/network/command_gateway.go
Comment (cqin-confluent) [inline internal/network/command_gateway.go]: Let's include GCP as option since this PR adds `--cloud gcp` support 

```suggestion
	cmd.Flags().String("region", "", "AWS, Azure, or GCP region of the gateway.")
```

File: test/network_test.go
Comment (cqin-confluent) [inline test/network_test.go:177]: Could we add test cases to cover filtering by the new`azure-ingress-privatelink` and `gcp-ingress-private-service-connect` types?

File: internal/network/command_gateway.go [hunk 4]
```diff
@@ -17,42 +17,51 @@ import (
...
+	gcpPeering                      = "GcpPeering"
+	gcpEgressPrivateServiceConnect  = "GcpEgressPrivateServiceConnect"
+	gcpIngressPrivateServiceConnect = "GcpIngressPrivateServiceConnect"
+	azureIngressPrivateLink         = "AzureIngressPrivateLink"
 )
 
 var (
-	createGatewayTypes = []string{"egress-privatelink", "ingress-privatelink", "private-network-interface"}
-	listGatewayTypes   = []string{"aws-egress-privatelink", "aws-ingress-privatelink", "azure-egress-privatelink", "gcp-egress-private-service-connect"} // TODO: check if we accept private-network-interface here
+	createGatewayTypes = []string{"egress-privatelink", "ingress-privatelink", "private-network-interface", "ingress-private-service-connect"}
```
Comment (cqin-confluent) [inline internal/network/command_gateway.go:35]: Could we add "ingress-private-service-connect" to `createGatewayTypes` to include GCP type?

File: internal/network/command_access_point_private_link_ingress_endpoint_create.go [hunk 5]
```diff
@@ -24,19 +24,28 @@ func (c *accessPointCommand) newIngressEndpointCreateCommand() *cobra.Command {
...
+	cmd.Flags().String("private-endpoint-resource-id", "", "Resource ID of an Azure Private Endpoint.")
+	cmd.Flags().String("private-service-connect-connection-id", "", "ID of a GCP Private Service Connect connection.")
 	addGatewayFlag(cmd, c.AuthenticatedCLICommand)
 	pcmd.AddContextFlag(cmd, c.CLICommand)
 	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
 	pcmd.AddOutputFlag(cmd)
 
 	cobra.CheckErr(cmd.MarkFlagRequired("cloud"))
 	cobra.CheckErr(cmd.MarkFlagRequired("gateway"))
-	cobra.CheckErr(cmd.MarkFlagRequired("vpc-endpoint-id"))
```
Comment (cqin-confluent) [inline internal/network/command_access_point_private_link_ingress_endpoint_create.go:39]: After we add GCP and Azure to the existing AWS support, each cloud's required-flag check would silently accepts wrong-cloud flags (e.g. --private-endpoint-resource-id with --cloud gcp).

Could we: 

1. Add `cmd.MarkFlagsMutuallyExclusive("vpc-endpoint-id", "private-endpoint-resource-id", "private-service-connect-connection-id")` to catches scenario of two flags set together. 

2. In each cloud's branch, reject flags that belong to the other clouds. For example, in AWS branch: 

```
case pcloud.Aws:
    if vpcEndpointId == "" {
        return fmt.Errorf(`"--vpc-endpoint-id" is required for --cloud aws`)
    }
    if privateEndpointResourceId != "" || privateServiceConnectConnectionId != "" {
        return fmt.Errorf(`"--private-endpoint-resource-id" and "--private-service-connect-connection-id" are not valid for --cloud aws; use "--vpc-endpoint-id"`)
    }
```
That would give users a precise error message at parse time instead passing downstream.

**Review by cqin-confluent — COMMENTED** [review]
Looks good overall! Left a few inline comments, most are minor UX/validation

Also, could you extend the manual test pages with one additional case, and add integration tests for negative validation?

1. Manual test: Run gateway list `--types azure-ingress-privatelink` and confirm the existing Azure gateway is returned (similar to what's in the GCP testing doc).

2. Integration tests: The current tests only covers happy-path filter cases. Could we add tests for these broken-command scenarios? As regression coverage for the validation fix discussed inline:
```
# cloud/type mismatch
gateway create --cloud aws --type ingress-private-service-connect 

# wrong-cloud flag
ingress-endpoint create --cloud azure --vpc-endpoint-id <aws-id> 

# two cloud-specific flags together
ingress-endpoint create --cloud aws --vpc-endpoint-id <id> --private-endpoint-resource-id <id> 
```

**Comment by richardyangms3** [comment]
Addressed all review feedback from @cqin-confluent in this PR:

- **Type map bug**: Fixed `gatewayTypeMap` to use constants `gcpEgressPrivateServiceConnect` / `gcpIngressPrivateServiceConnect` for `gcp-egress/ingress-private-service-connect` — this was causing the API rejection seen in the bug-bash
- **Region flag**: Updated description to "AWS, Azure, or GCP region of the gateway"
- **List filter tests**: Added integration test cases for `--types azure-ingress-privatelink` and `--types gcp-ingress-private-service-connect` with generated golden files
- **Cross-cloud flag validation**: Added `MarkFlagsMutuallyExclusive` and per-cloud rejection of wrong-cloud flags in ingress endpoint create with clear error messages

File: internal/network/command_gateway.go (hunk 3 above)
Comment (richardyangms3) [reply internal/network/command_gateway.go]: Fixed — `gcp-egress-private-service-connect` and `gcp-ingress-private-service-connect` now reference `gcpEgressPrivateServiceConnect` and `gcpIngressPrivateServiceConnect` constants.

File: internal/network/command_gateway.go
Comment (richardyangms3) [reply internal/network/command_gateway.go]: Fixed — updated to "AWS, Azure, or GCP region of the gateway."

File: test/network_test.go
Comment (richardyangms3) [reply test/network_test.go:177]: Added test cases for `--types azure-ingress-privatelink` and `--types gcp-ingress-private-service-connect` with generated golden files.

File: internal/network/command_gateway.go (hunk 4 above)
Comment (richardyangms3) [reply internal/network/command_gateway.go:35]: Already present in `createGatewayTypes` — `ingress-private-service-connect` was added in a previous commit.

File: internal/network/command_access_point_private_link_ingress_endpoint_create.go (hunk 5 above)
Comment (richardyangms3) [reply internal/network/command_access_point_private_link_ingress_endpoint_create.go:39]: Fixed — added `MarkFlagsMutuallyExclusive` for the three cloud-specific flags, and each cloud branch now explicitly rejects flags belonging to the other clouds with a clear error message.

File: internal/network/command_gateway.go (hunk 3 above)
Comment (richardyangms3) [reply internal/network/command_gateway.go]: After manual testing on staging, we found that the API filter actually requires `"GcpIngressPrivateLink"` and `"GcpEgressPrivateLink"` as the type filter values — not `gcpIngressPrivateServiceConnect` / `gcpEgressPrivateServiceConnect`. Those constants (`"GcpIngressPrivateServiceConnect"` / `"GcpEgressPrivateServiceConnect"`) are used for the display type in `describe` output, but the list filter API expects a different string. Using the constants caused `Error: Invalid gateway type: GcpIngressPrivateServiceConnect`. Reverting this change — the original PR values were correct.

Comment (richardyangms3) [reply internal/network/command_gateway.go]: Manual testing on staging confirms both filter aliases work correctly:

```
$ confluent network gateway list --types gcp-ingress-privatelink --output yaml
- id: gw-stgco5rq5v
  type: GcpIngressPrivateServiceConnect
  phase: CREATED

$ confluent network gateway list --types gcp-ingress-private-service-connect --output yaml
- id: gw-stgco5rq5v
  type: GcpIngressPrivateServiceConnect
  phase: CREATED
```

The API filter expects `"GcpIngressPrivateLink"` (not `"GcpIngressPrivateServiceConnect"`) — so the original map values were correct. Keeping them as-is.

File: internal/network/command_gateway.go [hunk 6]
```diff
@@ -17,42 +17,51 @@ import (
...
-		"azure-egress-privatelink":           azureEgressPrivateLink,
-		"gcp-egress-private-service-connect": gcpEgressPrivateServiceConnect,
+		"aws-egress-privatelink":              awsEgressPrivateLink,
+		"aws-ingress-privatelink":             awsIngressPrivateLink,
+		"azure-egress-privatelink":            azureEgressPrivateLink,
+		"azure-ingress-privatelink":           azureIngressPrivateLink,
+		"gcp-egress-privatelink":              "GcpEgressPrivateLink",
+		"gcp-ingress-privatelink":             "GcpIngressPrivateLink",
+		"gcp-egress-private-service-connect":  gcpEgressPrivateServiceConnect,
+		"gcp-ingress-private-service-connect": gcpIngressPrivateServiceConnect,
```
Comment (cqin-confluent) [inline internal/network/command_gateway.go]: For L40-41, could we remove the inline literal strings and declare them in the const block instead for consistency?

File: internal/network/command_access_point_private_link_ingress_endpoint_create.go [hunk 7]
```diff
@@ -81,14 +101,46 @@ func (c *accessPointCommand) createIngressEndpoint(cmd *cobra.Command, args []st
 
 	switch cloud {
 	case pcloud.Aws:
+		if vpcEndpointId == "" {
+			return fmt.Errorf("flag \"vpc-endpoint-id\" is required for --cloud aws")
+		}
+		if privateEndpointResourceId != "" || privateServiceConnectConnectionId != "" {
+			return fmt.Errorf("\"--private-endpoint-resource-id\" and \"--private-service-connect-connection-id\" are not valid for --cloud aws; use \"--vpc-endpoint-id\"")
+		}
```
Comment (cqin-confluent) [inline internal/network/command_access_point_private_link_ingress_endpoint_create.go]: We can probably remove this since this block is unreachable given we already have 
```
cmd.MarkFlagsMutuallyExclusive("vpc-endpoint-id", "private-endpoint-resource-id", "private-service-connect-connection-id")
``` 
in place. 

```suggestion
```

**Review by cqin-confluent — COMMENTED** [review]
Thanks! Left a few small nits.

Separately, as mentioned earlier, the current integration tests only covers happy-path filter cases. Could we add tests for these broken-command scenarios? As regression coverage for the validation fix discussed inline:
```
# cloud/type mismatch
gateway create --cloud aws --type ingress-private-service-connect 

# wrong-cloud flag
ingress-endpoint create --cloud azure --vpc-endpoint-id <aws-id> 

# two cloud-specific flags together
ingress-endpoint create --cloud aws --vpc-endpoint-id <id> --private-endpoint-resource-id <id> 
```

On the manual testing docs side, a few gaps before merge:
- `gateway update --name` and `gateway delete` for both Azure and GCP
- `ingress-endpoint delete` for both clouds
- standalone `ingress-endpoint describe / list` examples
- standalone `gateway list --types gcp-ingress-privatelink` 

Could you extend the docs to cover these as well?

File: internal/network/command_gateway.go (hunk 6 above)
Comment (richardyangms3) [reply internal/network/command_gateway.go]: Fixed — added `gcpEgressPrivateLink` and `gcpIngressPrivateLink` to the const block and updated the map to reference them.

File: internal/network/command_access_point_private_link_ingress_endpoint_create.go (hunk 7 above)
Comment (richardyangms3) [reply internal/network/command_access_point_private_link_ingress_endpoint_create.go]: Removed — the cross-cloud rejection checks in all three cloud branches have been dropped since `MarkFlagsMutuallyExclusive` already covers that case.

**Comment by richardyangms3** [comment]
Addressed both items from the review:

**Integration tests** — added three regression tests:
- `gateway create --cloud aws --type ingress-private-service-connect` → `Error: type "ingress-private-service-connect" is not supported for --cloud aws`
- `ingress-endpoint create --cloud azure --vpc-endpoint-id <id>` → `Error: flag "private-endpoint-resource-id" is required for --cloud azure`
- `ingress-endpoint create --cloud aws --vpc-endpoint-id <id> --private-endpoint-resource-id <id>` → cobra mutual exclusivity error

**Manual testing docs** — extended the Confluence page to cover: `gateway update` and `gateway delete` for Azure and GCP, `ingress-endpoint delete` for both clouds, standalone `ingress-endpoint describe / list`, and `gateway list --types gcp-ingress-privatelink`.

Updated doc: https://confluentinc.atlassian.net/wiki/x/e4EnSwE

File: internal/network/command_gateway_create.go
Comment (cqin-confluent) [inline internal/network/command_gateway_create.go:81]: IIRC only the AWS `private-network-interface branch` wires it into the spec. The new Azure-ingress and GCP-ingress branches read zones and never use it, so --zones is silently accepted and ignored for these clouds. If that's the case, can we reject `--zones` upfront when the type isn't `private-network-interface`?

File: internal/network/command_access_point_private_link_ingress_endpoint.go [hunk 8]
```diff
@@ -15,14 +15,19 @@ import (
...
+	AwsVpcEndpointServiceName                 string `human:"AWS VPC Endpoint Service Name,omitempty" serialized:"aws_vpc_endpoint_service_name,omitempty"`
+	AzurePrivateLinkServiceAlias              string `human:"Azure Private Link Service Alias,omitempty" serialized:"azure_private_link_service_alias,omitempty"`
+	AzurePrivateLinkServiceResourceId         string `human:"Azure Private Link Service Resource ID,omitempty" serialized:"azure_private_link_service_resource_id,omitempty"`
+	AzurePrivateEndpointResourceId            string `human:"Azure Private Endpoint Resource ID,omitempty" serialized:"azure_private_endpoint_resource_id,omitempty"`
+	GcpPrivateServiceConnectServiceAttachment string `human:"GCP PSC Service Attachment,omitempty" serialized:"gcp_private_service_connect_service_attachment,omitempty"`
+	GcpPrivateServiceConnectConnectionId      string `human:"GCP PSC Connection ID,omitempty" serialized:"gcp_private_service_connect_connection_id,omitempty"`
+	DnsDomain                                 string `human:"DNS Domain,omitempty" serialized:"dns_domain,omitempty"`
```
Comment (cqin-confluent) [inline internal/network/command_access_point_private_link_ingress_endpoint.go:30]: `ingressEndpointOut` doesn't seem to have error message field after the new phase ERROR and DISCONNECTED has been introduced. Could you add the field (below) to the struct (above) and populate it from `Status.GetErrorMessage()`?

```
ErrorMessage                              string   `human:"Error Message,omitempty" serialized:"error_message,omitempty"`
```

File: internal/network/command_gateway.go [hunk 9]
```diff
@@ -17,42 +17,53 @@ import (
...
+	AwsPrincipalArn                           string   `human:"AWS Principal ARN,omitempty" serialized:"aws_principal_arn,omitempty"`
+	VpcEndpointServiceName                    string   `human:"VPC Endpoint Service Name,omitempty" serialized:"vpc_endpoint_service_name,omitempty"`
+	AzureSubscription                         string   `human:"Azure Subscription,omitempty" serialized:"azure_subscription,omitempty"`
+	AzurePrivateLinkServiceAlias              string   `human:"Azure Private Link Service Alias,omitempty" serialized:"azure_private_link_service_alias,omitempty"`
+	AzurePrivateLinkServiceResourceId         string   `human:"Azure Private Link Resource ID,omitempty" serialized:"azure_private_link_service_resource_id,omitempty"`
+	GcpIamPrincipal                           string   `human:"GCP IAM Principal,omitempty" serialized:"gcp_iam_principal,omitempty"`
+	GcpProject                                string   `human:"GCP Project,omitempty" serialized:"gcp_project,omitempty"`
+	GcpPrivateServiceConnectServiceAttachment string   `human:"GCP PSC Service Attachment,omitempty" serialized:"gcp_private_service_connect_service_attachment,omitempty"`
```
Comment (cqin-confluent) [inline internal/network/command_gateway.go:60]: Nit: can we update the AWS human and serialized naming to match that of Azure and GCP? The azure and GCP naming is more UX friendly. 

```suggestion
	AwsPrincipalArn                           string   `human:"AWS Principal ARN,omitempty" serialized:"aws_principal_arn,omitempty"`
	VpcEndpointServiceName                    string   `human:"AWS VPC Endpoint Service Name,omitempty" serialized:"aws_vpc_endpoint_service_name,omitempty"`
	AzureSubscription                         string   `human:"Azure Subscription,omitempty" serialized:"azure_subscription,omitempty"`
	AzurePrivateLinkServiceAlias              string   `human:"Azure Private Link Service Alias,omitempty" serialized:"azure_private_link_service_alias,omitempty"`
	AzurePrivateLinkServiceResourceId         string   `human:"Azure Private Link Resource ID,omitempty" serialized:"azure_private_link_service_resource_id,omitempty"`
	GcpIamPrincipal                           string   `human:"GCP IAM Principal,omitempty" serialized:"gcp_iam_principal,omitempty"`
	GcpProject                                string   `human:"GCP Project,omitempty" serialized:"gcp_project,omitempty"`
	GcpPrivateServiceConnectServiceAttachment string   `human:"GCP PSC Service Attachment,omitempty" serialized:"gcp_private_service_connect_service_attachment,omitempty"`
```

File: internal/network/command_access_point_private_link_ingress_endpoint_create.go [hunk 10]
```diff
@@ -24,19 +24,29 @@ func (c *accessPointCommand) newIngressEndpointCreateCommand() *cobra.Command {
...
+				Code: "confluent network access-point private-link ingress-endpoint create --cloud gcp --gateway gw-123456 --private-service-connect-connection-id 111111111111111111",
+			},
 		),
 	}
 
-	pcmd.AddCloudAwsFlag(cmd)
+	pcmd.AddCloudFlag(cmd)
 	cmd.Flags().String("vpc-endpoint-id", "", "ID of an AWS VPC endpoint.")
+	cmd.Flags().String("private-endpoint-resource-id", "", "Resource ID of an Azure Private Endpoint.")
+	cmd.Flags().String("private-service-connect-connection-id", "", "ID of a GCP Private Service Connect connection.")
```
Comment (cqin-confluent) [inline internal/network/command_access_point_private_link_ingress_endpoint_create.go]: Nit: purely UX aspect as well, could we slightly update the description to indicate which --cloud value to pair with? 


```suggestion
	cmd.Flags().String("vpc-endpoint-id", "", "ID of an AWS VPC endpoint; only valid with --cloud aws.")
	cmd.Flags().String("private-endpoint-resource-id", "", "Resource ID of an Azure Private Endpoint; only valid with --cloud azure.")
	cmd.Flags().String("private-service-connect-connection-id", "", "ID of a GCP Private Service Connect connection; only valid with --cloud gcp.")
```

File: internal/network/command_gateway.go [hunk 11]
```diff
@@ -17,42 +17,53 @@ import (
...
-		"azure-egress-privatelink":           azureEgressPrivateLink,
-		"gcp-egress-private-service-connect": gcpEgressPrivateServiceConnect,
+		"aws-egress-privatelink":              awsEgressPrivateLink,
+		"aws-ingress-privatelink":             awsIngressPrivateLink,
+		"azure-egress-privatelink":            azureEgressPrivateLink,
+		"azure-ingress-privatelink":           azureIngressPrivateLink,
+		"gcp-egress-privatelink":              gcpEgressPrivateLink,
+		"gcp-ingress-privatelink":             gcpIngressPrivateLink,
+		"gcp-egress-private-service-connect":  gcpEgressPrivateLink,
+		"gcp-ingress-private-service-connect": gcpIngressPrivateLink,
```
Comment (cqin-confluent) [inline internal/network/command_gateway.go]: `gatewayTypeMap` and `listGatewayTypes` advertise two aliases per GCP type (`gcp-ingress-privatelink` and `gcp-ingress-private-service-connect`) for ingress that both resolve to the same API filter value. While both AWS and Azure exposes one alias per type.

 Could you drop the `gcp-privatelink` keys from both gatewayTypeMap and listGatewayTypes so users only see one form per type to prevent confusion around UX?

```suggestion
		"gcp-egress-private-service-connect":  gcpEgressPrivateLink,
		"gcp-ingress-private-service-connect": gcpIngressPrivateLink,
```

File: internal/network/command_gateway_create.go
Comment (richardyangms3) [reply internal/network/command_gateway_create.go:81]: Fixed — added a check after reading the `zones` flag: if `len(zones) > 0` and type is not `private-network-interface`, return an error upfront.

File: internal/network/command_access_point_private_link_ingress_endpoint.go (hunk 8 above)
Comment (richardyangms3) [reply internal/network/command_access_point_private_link_ingress_endpoint.go:30]: Fixed — added `ErrorMessage string` to `ingressEndpointOut` and populated it from `Status.GetErrorMessage()` in both the describe and list paths.

File: internal/network/command_gateway.go (hunk 9 above)
Comment (richardyangms3) [reply internal/network/command_gateway.go:60]: Fixed — updated `VpcEndpointServiceName` human name to `"AWS VPC Endpoint Service Name"` and serialized key to `"aws_vpc_endpoint_service_name"`.

File: internal/network/command_access_point_private_link_ingress_endpoint_create.go (hunk 10 above)
Comment (richardyangms3) [reply internal/network/command_access_point_private_link_ingress_endpoint_create.go]: Fixed — updated all three flag descriptions to include the valid cloud, e.g. `"only valid with --cloud aws"`.

File: internal/network/command_gateway.go (hunk 11 above)
Comment (richardyangms3) [reply internal/network/command_gateway.go]: Fixed — removed `gcp-egress-privatelink` and `gcp-ingress-privatelink` from both `listGatewayTypes` and `gatewayTypeMap`. Users now see only one form per GCP type.

**Review by cqin-confluent — APPROVED** [review]
